src/sys/fuchsia/handles.rs

```diff
@@ -0,0 +1,66 @@
+use {io, poll, Evented, Ready, Poll, PollOpt, Token};
+use magenta::HandleBase;
+use std::sync::Mutex;
+
+/// Wrapper for registering a `HandleBase` type with mio.
+#[derive(Debug)]
+pub struct EventedHandle<T> where T: HandleBase {
```

Comment:
Alternatively, we could just convert the type into its inner `Handle` to store in `EventedHandle` and make it the responsibility of our wrapper crate to do the conversion. However, I think this approach is strictly more flexible-- if we go with this approach, in our wrapper crate we could just only ever use `PollEvented<EventedHandle<Handle>>` if that turns out to be a better approach (e.g. it could reduce the amount of code generated by monomorphization).